### DISCUSSION TITLE
"a good BTRFS structure in your filesystem"

### DISCUSSION THREAD

**Discussion by iconoclasthero** [egara/buttermanager#57] [General]
A what now?  I'm not sure what it is I'm supposed to take from a link to a very broad setup document for a distro I don't run.

Can you spell this out, or better yet, make things flexible enough so that one can backup any btrfs volume they have?  It is anathema to me to put /home and / on the same partition so if that's what is defined as "good btrfs structure," then thanks, no.

I neither understand nor care why this arrangement is thought desirable and whatever potential benefits derived from such a layout, my cons outnumber my pros for such an format.

At the end of the day, this is the first line of substance for your app's readme and yet buttermanager does nothing to explain what this is or why it is important and why I should care at all.  I just want to back up the volumes I want to back up and restore them when I want.  

The fact that I can choose to launch images from grub is a feature I find useful and apparently timeshift won't deign to backup my /home btrfs volume since it is not a subvolume of the main system so eventually, I'll need to replace that with something.

**Comment by egara** [comment]
Hi @iconoclasthero 

First of all, of course, we can discuss anything related to Buttermanager here, but please, let's always use good manners when doing so. That way, we can all enrich ourselves and reach a conclusion together.

The first thing I miss in your message is a little bit of context. Could you explain what is the problem you have?

**Comment by iconoclasthero** [comment]
So as directly stated as possible:
I do not intend to use the example filesystem/subvolume layout in the [linked document](https://github.com/egara/arch-btrfs-installation) from the buttermanager readme particularly vis-a-vis `/` and `/home`.  Why such an arrangement is considered good is irrelevant to me since I won't be doing it that way regardless of any potential nose cutting which could spite my face.  (I try to be clear about that up front since we're not going to be getting into a "you need to do it this way because that's the right way to do it" back and forth; please don't take it personally.)

You give exactly one example that has two subvolumes from a single /dev/sda3 partition:
```
sda3 (Volume)
|
|
- _active (Subvolume)
|    |
|    - rootvol (Subvolume - It will be the current /)
|    - homevol (Subvolume - it will be the current /home)
|
- _snapshots (Subvolume -  It will contain all the snapshots which are subvolumes too)
```
(FYI calamares set up the btrfs system I installed also included a `- cache` subvolume which I also do not use.)

Timeshift apparently will not do anything with the separate btrfs /home and /library partitions I have so right off the bat, I have a problem with it and it will eventually need to be replaced or augmented with something that will take snaps of my other file systems.  It has no grub integration.

buttermanager has the aforementioned caveat that is poorly elucidated in terms of what other arrangements are and are not acceptable.  It has grub integration from btrfs-grub that works without me needing to know how.  

I ran into butter today on [reddit](https://www.reddit.com/r/Fedora/comments/wuotk0/butter_yet_another_btrfs_snapshot_manager_and/) and it the OP specifically stated that, "Some ask for a very specific subvolume layout, other has a learning curve that I am not willing to commit to (yes, so I spent months creating this instead of reading probably 1-2 hours of documentation). Butter should handle most if not all Btrfs subvolume layouts (including Fedora's default layout). It even supports having multiple Btrfs filesystems."  (NB: I'm the only one who has a problem with btrfs snapshot managers that are limited to prescribes structures.)

I want a btrfs manager that
- is agnostic to the filesystem/subvolume layout/hierarchy
- is integrated with grub-btrfs so I can select snaps at boot.
- will take snaps automatically so that it actually gets done.

Combining butter and buttermanger would give me a GTK4/GNOME app that's agnostic to subvolume hierarchy and is integrated with grub-btrfs to select root images at boot. (I do not expect that grub-btrfs is going to manage other things besides what is mounted at /, but if it could do more than that, so much the better.)

So is there some way that buttermanager can handle my rhino `/` as sdb6/@ while still being able to take snaps of a home btrfs-formatted partition that is on /dev/sdb9 and a /library partition at /dev/sda?

What other arrangements can buttermanager handle other than the n=1 example you give?

**Comment by egara** [comment]
The layout I suggest [here](https://github.com/egara/arch-btrfs-installation) is only a **suggestion**. ButterManager can work with any layout you want. If you want to take advantage of GRUB integration via grub-btrfs, it is needed to include **/boot** into the subvolume where **/** is located. That's all. And it is important because when you rollback to a previous / snapshot, you want the content of the /boot rolled back accordingly (imagine that you have updated your kernel. The images generated by mkinitcpio and used to init the system when you boot need to be in sync with the kernel you have).

The other important thing is to mount the whole BTRFS volume in /mnt for example. This way, you will be able to tell ButterManager where to store all the snapshots created. I think you haven't taken a look at the Wiki of ButterManager [here](https://github.com/egara/buttermanager/wiki/Documentation).

As I told you, the layout is not important. Just configure ButterManager with the subvolumes you want to manage and provide a subvolume to store all the snapshots created.
